src/test/uint256_tests.cpp

```diff
@@ -328,6 +328,15 @@ BOOST_AUTO_TEST_CASE(parse)
         BOOST_CHECK_EQUAL(uint256S(" 0x").GetHex(), s_0);
         BOOST_CHECK_EQUAL(uint256S(" ").GetHex(), s_0);
     }
+    {
+        // Make sure FromHex respects string_view length and stops reading at
+        // end of the substring.
+        auto chars_68{"0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123"};
+        std::string_view chars_64{chars_68, 64};
+        BOOST_CHECK(!uint256::FromHex(std::string_view(chars_68, 63))); // too short
+        BOOST_CHECK_EQUAL(uint256::FromHex(std::string_view(chars_68, 64)).value().ToString(), chars_64);
+        BOOST_CHECK(!uint256::FromHex(std::string_view(chars_68, 65))); // too long
+    }
```

Comment:
Good to see the former `TxidFromString()` test resurrected with even more checks!

I noticed there are no tests for `FromHex` returning `nullopt` for invalid characters, care to add?

Maybe also mix up the hex case in `chars_68`?
`0123456789abcdef0123456789ABCDEF0123456789abcdef0123456789ABCDEF0123`

Reply:
I've 17ebf1ac133d26676ba7ae3e40042fdf389d146f with a bunch more unit test coverage for `uint256::FromHex()`  and methods that wrap it, including invalid characters, `0x` prefix, and mixed case. Additionally, I've also added a fuzz target in d10e351d3df93f60a359d8a2c0bc9fbd175395f8